### PR TITLE
Add Aura Recon Companion dev-container scaffold

### PR DESCRIPTION
**Pull request by MVPuknowme**
## Summary

Adds the first development scaffold for **Aura Recon Companion**: a SwiftUI iPhone client source tree and a bounded FastAPI service that can run in GitHub Codespaces.

## Included

- one-click Codespaces/dev-container configuration
- SwiftUI app source and XcodeGen project definition
- authorization confirmation before any target request
- hostname-only input validation
- blocking for private, loopback, link-local, reserved, and other non-global targets
- bounded DNS record collection
- optional single HTTPS metadata request
- SHA-256 evidence digest generation
- PowerShell startup script
- backend tests and path-scoped GitHub Actions workflow

## Security boundaries

- no exploitation, credential testing, path brute forcing, arbitrary terminal execution, or private-network enumeration
- no claim of full SpiderFoot module parity in this first scaffold
- SpiderFoot-scale provider integrations remain a separately sandboxed follow-up
- no production deployment or Vercel configuration changes

## Distribution boundary

The Codespaces link is available immediately for development. A physical-iPhone download link still requires Apple signing, Xcode archive, App Store Connect/TestFlight configuration, and Apple review where applicable.

## Validation

The branch includes a Python 3.12 GitHub Actions test job. Local validation from the assistant container was blocked because that container could not resolve GitHub; the PR checks are the source of truth.

### CODE REVIEW

File: apps/aura-recon-companion/backend/app/scanner.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+
+def _web_metadata_findings(target: str) -> list[Finding]:
+    url = f"https://{target}/"
+    with httpx.Client(
+        timeout=httpx.Timeout(8.0, connect=5.0),
+        follow_redirects=False,
+        headers={"User-Agent": "Aura-Recon-Companion/0.1 authorized-metadata-check"},
+    ) as client:
+        response = client.head(url)
```
Comment (MVPuknowme) [inline apps/aura-recon-companion/backend/app/scanner.py]: Blocking SSRF issue: the hostname was validated via `socket.getaddrinfo()` above, but `client.head(url)` performs a new DNS lookup. A rebinding hostname can be public during validation and resolve to a private/loopback/link-local address here. Pin the connection to a previously validated global IP while preserving the original hostname for TLS SNI/Host, or keep this profile disabled.

File: apps/aura-recon-companion/backend/tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    )
+    assert response.status_code == 403
+
+
+def test_rejects_url_in_target() -> None:
+    response = client.post(
+        "/api/v1/scans",
+        json={"target": "https://example.com/path", "authorized": True, "profile": "dns-passive"},
+    )
+    assert response.status_code == 422
```
Comment (MVPuknowme) [inline apps/aura-recon-companion/backend/tests/test_api.py:31]: Please add mocked scanner-level tests for the claimed boundary: reject private and mixed global/private resolutions, prevent a second DNS lookup/rebinding on HTTPS, handle DNS/HTTPS timeouts, and cap DNS/header output. These current tests never reach `run_scan()` with an authorized, syntactically valid target.

**Review by MVPuknowme — COMMENTED** [review]
Review result: **changes needed before merge**.

Blocking:

1. The HTTPS metadata path is vulnerable to DNS rebinding/TOCTOU. `_validated_public_addresses()` validates one resolution, then `httpx` resolves the hostname again when making the request. An attacker-controlled hostname can return a global address for validation and a loopback/private/link-local address for the second lookup, bypassing the PR's stated SSRF boundary. Pin the outbound TLS connection to one of the already-validated global IPs while retaining the original hostname for SNI and the Host header, or remove/disable `web-metadata` until that transport exists.
2. Add regression coverage for the actual scanner boundary: public resolution success, non-global and mixed-address rejection, DNS rebinding, DNS timeout/failure, HTTPS timeout, and bounded DNS/header output. The current three API tests confirm health, the authorization boolean, and URL-shaped input only; they do not execute the scanner's security-critical path.

Verified:

- PR is open, draft, and mergeable.
- `Aura Recon Companion`, `SKYGRID IOC Watch`, and `SKYGRID Controlled Pilot Verification` GitHub Actions runs all succeeded for `31b38f0`.
- No pre-existing review comments or unresolved threads.
- Five Vercel status contexts fail because deployment is account-blocked. This PR declares no production deployment changes, but those contexts remain an external merge blocker if branch protection requires them.

Before any Codespaces port is shared publicly or the backend is deployed, add real service authentication and per-client rate limits; the `authorized` request boolean is user attestation, not access control.

GitHub does not permit the connected PR author account to submit `REQUEST_CHANGES` on its own PR, so this is recorded as a blocking comment rather than a formal change-request state.
